src/main/java/de/hpi/bpt/chimera/execution/Case.java

```diff
@@ -76,15 +80,57 @@ public FragmentInstance addFragmentInstance(Fragment fragment) {
 		return null;
 	}
 
-	public void removeFragmentInstance(FragmentInstance fragmentInstance) {
+	/**
+	 * Instantiate a specific fragment and enable it if the instantiation was
+	 * successful.
+	 * 
+	 * @param fragment
+	 *            - that will be instantiated.
+	 * @see #instantiateFragment(Fragment) instantiateFragment
+	 */
+	public synchronized void instantiateFragmentAndEnableInstance(Fragment fragment) {
+		FragmentInstance fragmentInstance = instantiateFragment(fragment);
+		if (fragmentInstance != null) {
+			fragmentInstance.enable();
+		}
+	}
+
+	/**
+	 * Check whether a new instantiation of a fragment can be created. Therefore
+	 * it must be defined that the fragment has no bound or there are exists
+	 * less instantiation than the defined limit.
+	 * 
+	 * @param fragment
+	 *            - to check
+	 * @return true if another instantiation is possible.
+	 * @see {@link Fragment}
+	 */
+	private boolean isInstantiable(Fragment fragment) {
+		if (!fragment.getHasBound()) {
+			return true;
+		}
+		long existingInstancesAmount = getFragmentInstances().values().stream()
```

Comment:
Is this not a expensive operation to always stream all fragment instances and test for equal? Could we not just have a static variable `numberOfInstances` in the `Fragment` class?

Reply:
If there are more and more instances of a fragment you are right.
I would not recommend a variable in the `Fragment` class because it would mix the model and the execution. In addition there could be more instantiations of one one fragment in different cases but I thought of the instantiation bound as a limit how often a fragment can be instantiated in *one* case.